src/QirRuntime/lib/QIR/callables.cpp

```diff
@@ -107,6 +96,60 @@ extern "C"
     }
 }
 
+/*==============================================================================
+    Implementation of QirTupleHeader
+==============================================================================*/
+int QirTupleHeader::AddRef()
+{
+    assert(refCount > 0);
+    return ++refCount;
+}
+
+int QirTupleHeader::Release()
+{
+    --refCount;
+    if (refCount == 0)
+    {
+        char* buffer = reinterpret_cast<char*>(this);
+        delete[] buffer;
+    }
+    return refCount;
+}
+
+QirTupleHeader* QirTupleHeader::Create(int size)
+{
+    assert(size >= 0);
+    char* buffer = new char[sizeof(QirTupleHeader) + size];

```

Comment:
Ensure buffer!=0

Reply:
If new[] fails to allocate it throws std::bad_alloc. There is a non-throwing version of operator new[] that would return a null pointer but I'm not sure it's worth the trouble to handle OOM situations ourselves rather than let the execution fail with the exception. Thoughts?